apps/application/pages/onboarding/mods/poap.tsx

```diff
@@ -10,118 +9,121 @@ const eventIdUsed = process.env.NEXT_PUBLIC_POAP_MODERATOR_EVENT_ID as string;
 const PoapModeratorsPage = ({
   eventInfo,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) => {
-  const address = useAddress();
-  const { claimPoap, poapScan, addressHasPoap, poapTokenId } =
+  const { claimPoap, poapScan, addressHasPoap, poapTokenId, connectedWallet } =
     useOnboardingContextState();
 
   useEffect(() => {
-    if (address) {
-      poapScan(address, eventIdUsed);
+    if (connectedWallet !== "" && addressHasPoap === false) {
+      poapScan(connectedWallet, eventIdUsed);
     }
-  }, [address]);
+  }, []);
 
   const handleClaimPoap = async () => {
-    if (address) {
-      claimPoap(address, eventIdUsed);
+    if (connectedWallet !== "" && addressHasPoap === false) {
+      claimPoap(connectedWallet, eventIdUsed);
```

Comment:
After minting the poap the screen keeps showing the claiming poap button and it don't update to the after minting screen; probably there should be another a **(1) loading modal during the minting** and **(2)poapScan just after claiming** the poap to renew the screen display:

https://github.com/zenbitETH/zengo/assets/57845110/a29d5296-9551-418f-9233-01d7d9280349

Reply:
@HabacucMX allright! will do it!